packages/shared/src/components/modals/CompanionModal.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { ReactElement, useMemo } from 'react';
+
+import { LazyModalCommonProps, Modal } from './common/Modal';
+import { Button } from '../buttons/Button';
+import UpvoteIcon from '../icons/Upvote';
+import CommentIcon from '../icons/Discuss';
+import LineIcon from '../icons/Line';
+import { IconSize } from '../Icon';
+import useMedia from '../../hooks/useMedia';
+import { mobileL, tablet } from '../../styles/media';
+import { cloudinary } from '../../lib/image';
+
+type Props = LazyModalCommonProps & {
+  url: string;
+  onReadArticleClick: (any) => void;
+  onActivateCompanion: () => void;
+};
+
+export default function CompanionModal({
+  onReadArticleClick,
+  onActivateCompanion,
+  url,
+  ...props
+}: Props): ReactElement {
+  const isTablet = !useMedia([tablet.replace('@media ', '')], [true], false);
+  const isMobile = !useMedia([mobileL.replace('@media ', '')], [true], false);
+
+  const size = useMemo(() => {
+    if (isMobile) return Modal.Size.XSmall;
+    if (isTablet) return Modal.Size.Small;
```

Comment:
Shouldn't our modals be responsive enough that we don't have to do this through JS? I could be wrong though.

Reply:
They are responsive but it does not seem like they match the design requirements by default -- they scale with the screen but are either too wide/don't leave any margins.